docs/fundamentals/annotations-overview.md

```diff
@@ -0,0 +1,265 @@
+
+Here's a simple example of how annotations are used:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var redis = builder.AddRedis("cache")
+    .WithCommand("clear-cache", "Clear Cache", OnClearCache)
+    .WithUrl("admin", "http://localhost:8080/admin");
+
+builder.Build().Run();
+```
+
+In this example:
+
+- `WithCommand` adds a `CommandAnnotation` that defines a custom command
+- `WithUrl` adds a `ResourceUrlAnnotation` that defines a custom URL
+
+## Built-in annotation types
+
+.NET Aspire includes several built-in annotation types for common scenarios:
+
+### EndpointAnnotation
+
+The <xref:Aspire.Hosting.ApplicationModel.EndpointAnnotation> defines network endpoints for resources. It contains information about ports, protocols, and endpoint configuration.
+
+```csharp
+var api = builder.AddProject<Projects.Api>("api")
+    .WithEndpoint(callback: endpoint =>
+    {
+        endpoint.Port = 5000;
+        endpoint.IsExternal = true;
+        endpoint.Protocol = Protocol.Tcp;
+        endpoint.Transport = "http";
+    });
+```
+
+### ResourceUrlAnnotation
+
+The <xref:Aspire.Hosting.ApplicationModel.ResourceUrlAnnotation> defines custom URLs that appear in the dashboard, often pointing to management interfaces or documentation.
+
+```csharp
+var database = builder.AddPostgres("postgres")
+    .WithUrl("admin", "https://localhost:5050");
+```
+
+### EnvironmentCallbackAnnotation
+
+The <xref:Aspire.Hosting.ApplicationModel.EnvironmentCallbackAnnotation> allows you to modify environment variables at runtime based on the state of other resources.
+
+```csharp
+// This is typically used internally by WithReference
+var app = builder.AddProject<Projects.MyApp>("app")
+    .WithReference(database);
+```
+
+### ContainerMountAnnotation
+
+The `ContainerMountAnnotation` defines volume mounts for containerized resources.
+
+```csharp
+var postgres = builder.AddPostgres("postgres")
+    .WithDataVolume(); // Adds a ContainerMountAnnotation
+```
+
+### CommandAnnotation
+
+The `CommandAnnotation` defines custom commands that can be executed from the dashboard.
+
+```csharp
+var redis = builder.AddRedis("cache")
+    .WithCommand("clear", "Clear Cache", OnClearCache);
+```
+
+## Creating custom annotations
+
+You can create custom annotations by implementing the <xref:Aspire.Hosting.ApplicationModel.IResourceAnnotation> interface. Here's how to create a custom annotation:
+
+### 1. Define the annotation class
+
+:::code source="snippets/annotations-overview/Program.cs" id="CustomConfigAnnotation":::
+
+### 2. Create extension methods
+
+Create extension methods to make your annotation easy to use:
+
+:::code source="snippets/annotations-overview/Program.cs" id="CustomConfigExtensions":::
+
+### 3. Use the custom annotation
+
+```csharp
+var api = builder.AddProject<Projects.Api>("api")
+    .WithCustomConfig("feature-flag", "enabled");
```

Comment:
I'm not a fan of the `CustomConfigAnnotation` example. Please propose another option.